internal/security/secretstore/kongadminapi.go

```diff
@@ -0,0 +1,241 @@
+	err = ioutil.WriteFile(k.paths.jwt, []byte(k.secrets.jwt.signed), 0644)
+	if err != nil {
+		return fmt.Errorf("%s Failed to write JWT to file %s: %w", k.prefixes.errText, k.paths.jwt, err)
+	}
+
+	return nil
+}
+
+// createJWT creates a JWT from a given private key and issuer payload value
+func (k *KongAdminAPI) createJWT(privateKey string, issuer string) (signedToken string, err error) {
+
+	// Setup JWT generation variables
+	now := time.Now().Unix()
+	duration, err := time.ParseDuration("1h")
+	if err != nil {
+		return "", fmt.Errorf("%s Could not parse JWT duration: %w", k.prefixes.errText, err)
+	}
+
+	// Sanity check - parse & check EC key
+	eckey, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKey))
+	if err == nil && eckey.Params().BitSize != 256 {
+		return "", fmt.Errorf("%s EC key bit size is incorrect (%d instead of 256)", k.prefixes.errText, eckey.Params().BitSize)
+	}
+	if err != nil {
+		return "", fmt.Errorf("%s Could not parse private key: %w", k.prefixes.errText, err)
+	}
+
+	// Create JWT
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, &jwt.StandardClaims{
+		Issuer:    issuer,
+		IssuedAt:  now,
+		NotBefore: now,
+		ExpiresAt: now + int64(duration.Seconds()),
+	})
+
+	// Save JWT to string return
+	signedToken, err = token.SignedString(eckey)
+	if err != nil {
+		return "", fmt.Errorf("%s Could not sign JWT: %w", k.prefixes.errText, err)
+	}
+
+	return
+}
+
+// clearSecrets clears the key structs by setting them to empty struct values
+func (k *KongAdminAPI) clearSecrets() {
+	k.secrets = KongAdminAPISecrets{}
+}
+
+// verifyPrefixes confirms that there are values set for the prefixes
+func (k *KongAdminAPI) verifyPrefixes() error {
+
+	base := "[build-kong-admin]"
+
+	if k.prefixes.cmdText == "" {
+		k.prefixes.cmdText = base
+	}
+
+	if k.prefixes.errText == "" {
+		k.prefixes.errText = base + "[error]"
+	}
+
+	if k.prefixes.infoText == "" {
+		k.prefixes.infoText = base + "[info]"
+	}
+
+	return nil
+}
+
+// verifyPaths confirms values are set for the paths, and that those paths exist
+func (k *KongAdminAPI) verifyPaths() error {
+
+	// Is the template path set?
+	if k.paths.template == "" {
+		return fmt.Errorf("%s Template file path (<KongAdminAPI>.paths.template) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the template path/file exist? It should be set.
+	if _, err := os.Stat(k.paths.template); os.IsNotExist(err) {
+		return fmt.Errorf("%s Template file path (<KongAdminAPI>.paths.template) does not exist - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Is the config file path set?
+	if k.paths.config == "" {
+		return fmt.Errorf("%s Kong configuration file path (<KongAdminAPI>.paths.config) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the config file path directory exist? If not, create it.
+	if _, err := os.Stat(path.Dir(k.paths.config)); os.IsNotExist(err) {
+		dirErr := os.MkdirAll(path.Dir(k.paths.config), 0777)
```

Comment:
do we need the directory permission to be 777?

Reply:
I don't know how else we could accomplish this. It's creating the configuration file as the "edgex_user" user and it's being read in the Kong routine as the "kong" user through volume mounts. If "everyone" can't read it, then the "kong" user won't be able to access it. 

For context - the contents of that configuration file are not dangerous in the event that someone else reads it. If someone is able to attach themselves to that volume mount - then we've got bigger problems than this config file.

Reply:
is this `chmod -R a+r /tmp/kong` doing for that or this is a separated directories?  As that seems to restrict the access already

Reply:
ok, 644 works for me